cluster/service/cluster_service.go

```diff
@@ -397,7 +397,7 @@ func (c clusterService) loadClusterByURL(ctx context.Context, clusterURL string)
 			// oops, something wrong happened, not just the cluster not found in the db
 			return nil, errs.Wrapf(err, "unable to load cluster")
 		}
-		return nil, errors.NewBadParameterError("cluster-url", fmt.Sprintf("cluster with requested url %s doesn't exist", clusterURL))
+		return nil, errors.NewNotFoundError("cluster", clusterURL)
```

Comment:
I see, `Not Found` already handled here [link](https://github.com/fabric8-services/fabric8-cluster/blob/a45d26c5309fdaf5ad86c9ee9a19bbe20e6990be/cluster/repository/cluster_repository.go#L117).  So we can avoid processing it here.

Even I see that we can optimize and can directly call `LoadClusterByURL` [link](https://github.com/fabric8-services/fabric8-cluster/blob/a45d26c5309fdaf5ad86c9ee9a19bbe20e6990be/cluster/repository/cluster_repository.go#L112) and no need of this wrapper method any more ??

Reply:
right, we could use that method, indeed. But I need to check if @alexeykazakov agrees with this change, first...